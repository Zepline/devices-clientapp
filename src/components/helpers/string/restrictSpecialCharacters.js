export default function restrictSpecialCharacters(event) {

    // eslint-disable-next-line
    const specialCharacters = new RegExp("^[a-zA-Z\-0-9 ]+$");
    const key = String.fromCharCode(!event.charCode ? event.which : event.charCode);

    if (!specialCharacters.test(key)) {

        event.preventDefault();
        return false;
    }
}