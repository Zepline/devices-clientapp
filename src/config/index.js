const defaults = {

    types: ["Windows Workstation", "Windows Server", "Mac"],
    deviceApiEndpoint: "http://localhost:3000/devices",
}

module.exports = { defaults };