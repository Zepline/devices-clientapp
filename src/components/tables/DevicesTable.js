import React, { useEffect } from 'react';
import { useTable, useFilters, useSortBy, usePagination } from 'react-table';
import restrictSpecialCharacters from '../helpers/string/restrictSpecialCharacters.js';
import { defaults } from '../../config/index';
import { useForm } from "react-hook-form";
import axios from 'axios';

function SelectColumnFilter({ column: { filterValue, setFilter, preFilteredRows, id } }) {

    const options = React.useMemo(() => {

        const options = new Set()

        preFilteredRows.forEach((row) => { options.add(row.values[id]) })
        return [...options.values()]

    }, [id, preFilteredRows])

    return (

        <select 
            className="devices-table-select-filter" 
            value={filterValue} 
            onChange={(e) => { 
                setFilter(e.target.value)
                localStorage.setItem('filter-setting', e.target.value)
            }}
        >
            <option value="">All</option>
            {options.map((option, i) => (<option key={i} value={option}>{option}</option>))}
        </select>
    )
}

function Table({ columns, data }) {
    
    if (!localStorage.getItem('filter-setting')) { localStorage.setItem('filter-setting', ""); }

    const {
        getTableProps,
        getTableBodyProps,
        headerGroups,
        prepareRow,
        page,
        canPreviousPage,
        canNextPage,
        pageOptions,
        pageCount,
        gotoPage,
        nextPage,
        previousPage,
        state: { pageIndex },
    } = useTable({ 
        columns, 
        data,
        initialState: {
            filters: [{
                id: 'type',
                value: localStorage.getItem('filter-setting')
            }]
        }
    }, useFilters, useSortBy, usePagination)

    return (

        <div>
            <div className="devices-table-container" {...getTableProps()}>
                <div className="devices-table-header-container">
                    {headerGroups.map((headerGroup) => (
                        <div className="devices-table-filter-row-container" {...headerGroup.getHeaderGroupProps()}>
                            {headerGroup.headers.map((column) => (
                                <div className="devices-table-filter-container" {...column.getHeaderProps()}>
                                    <div className="devices-table-filter-title">{column.render('filterName')}</div>
                                    <div>{column.canFilter ? column.render('Filter') : null}</div>
                                </div>
                            ))}
                        </div>
                    ))}
                </div>
                <div className="devices-table-body-container" {...getTableBodyProps()}>
                    {page.map((row) => {
                        prepareRow(row)

                        return (
                            <div className="devices-table-row-container" {...row.getRowProps()}>
                                {row.cells.map((cell) => { return (<div {...cell.getCellProps()}>{cell.render('Cell')} </div>) })}
                            </div>
                        )
                    })}
                </div>
            </div>
            <div className="pagination-container">
                <div className="pagination-buttons-container">
                    <div className="pagination-button" onClick={() => gotoPage(0)} disabled={!canPreviousPage}>{`<<`}</div>
                    <div className="pagination-button" onClick={() => previousPage()} disabled={!canPreviousPage}>{`<`}</div>
                    <div className="pagination-button" onClick={() => nextPage()} disabled={!canNextPage}>{`>`}</div>
                    <div className="pagination-button" onClick={() => gotoPage(pageCount - 1)} disabled={!canNextPage}>{`>>`}</div>
                </div>
                <div className="pagination-info-container">
                    <div className="page-of-text">Page{' '}{pageIndex + 1} of {pageOptions.length}</div>
                </div>
            </div>
        </div>
    )
}

function DevicesTable(input) {

    const assignedSortSetting = input.tableSortSetting;
    const [devicesData, setDevicesData] = React.useState([]);
    const [showEditDevicePopup, setShowEditDevicePopup] = React.useState(false);
    const [showDeleteDevicePopup, setShowDeleteDevicePopup] = React.useState(false);
    const [editData, setEditData] = React.useState([]);
    const [deleteData, setDeleteData] = React.useState([]);

    useEffect(() => {

        const abortController = new AbortController();

            axios.get(defaults.deviceApiEndpoint).then(response => {

                const responseData = response.data.map(device => ({
                    id: device.id,
                    system_name: device.system_name,
                    type: device.type,
                    hdd_capacity: device.hdd_capacity,
                    edit_data: device,
                    delete_data: device
                }));

                
                setDevicesData(responseData)

            }).catch(error => {

                setDevicesData([])
                console.log(error);
            })
       
        return () => abortController.abort();

    }, []);

    const { register, handleSubmit } = useForm();

    const onUpdateSubmit = (data) => {

        axios.put(`${defaults.deviceApiEndpoint}/${editData.id}`, data).then(() => {

            setShowEditDevicePopup(false);
            setEditData([]);
            setTimeout(function () { window.location.reload() }, 100);
        });
    };

    const onDeleteSubmit = () => {

        axios.delete(`${defaults.deviceApiEndpoint}/${deleteData.id}`).then(() => {

            setShowDeleteDevicePopup(false);
            setDeleteData([]);
            setTimeout(function () { window.location.reload() }, 100);
        });
    };

    function editClickHandler(data) {

        if (showEditDevicePopup === false) {

            setEditData(data)
            setShowEditDevicePopup(true)
        }
    }

    function deleteClickHandler(data) {

        if (showDeleteDevicePopup === false) {

            setDeleteData(data)
            setShowDeleteDevicePopup(true)
        }
    }

    function handleEditDeleteShow() {

        if (showEditDevicePopup === true) {

            return (

                <div className="devices-table-add-edit-popup-container-overlay">
                    <div className="devices-table-add-edit-popup-container">
                        <div className="devices-table-add-edit-popup-container-top">
                            <div className="devices-table-add-edit-popup-closer" onClick={() => setShowEditDevicePopup(false)}>â¤«</div>
                            <div className="devices-table-add-edit-popup-title">Edit Device</div>
                        </div>
                        <div className="devices-table-add-edit-popup-container-middle">
                            <div className="devices-table-add-edit-popup-container-device-details">
                                <div className="devices-table-add-edit-popup-details-title">Device Details:</div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">System Name:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{editData.system_name}</div>
                                </div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">Type:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{editData.type}</div>
                                </div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">HDD Capacity:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{editData.hdd_capacity}GB</div>
                                </div>
                            </div>
                            <form className="devices-table-add-edit-popup-form-container" onSubmit={handleSubmit(onUpdateSubmit)}>
                                <div className="devices-table-add-edit-popup-form-input-container">
                                    <div className="devices-table-add-edit-devices-popup-input-title">System Name *</div>
                                    <input className="devices-table-add-edit-devices-popup-input-data" 
                                    
                                        {...register("system_name", { required: true })} 
                                        type="text"
                                        maxLength="50"
                                        onKeyPress={(e) => restrictSpecialCharacters(e)}
                                        defaultValue={editData.system_name}
                                    />
                                </div>
                                <div className="devices-table-add-edit-popup-form-input-container">
                                    <div className="devices-table-add-edit-devices-popup-input-title">Type *</div>
                                    <select className="devices-table-add-edit-devices-popup-input-select" defaultValue={editData.type} {...register("type", { required: true })}>
                                        <option value="">Select Type</option>
                                        <option value="Windows Workstation">Windows Workstation</option>
                                        <option value="Windows Server">Windows Server</option>
                                        <option value="Mac">Mac</option>
                                    </select>
                                </div>
                                <div className="devices-table-add-edit-popup-form-input-container">
                                    <div className="devices-table-add-edit-devices-popup-input-title">HDD Capacity (GB)*</div>
                                    <input className="devices-table-add-edit-devices-popup-input-data" 
                                        defaultValue={editData.hdd_capacity}
                                        min="0" 
                                        type="number" 
                                        {...register("hdd_capacity", { required: true })} 
                                        />
                                </div>
                                <div className="devices-table-add-edit-popup-container-bottom">
                                    <input className="devices-table-add-edit-popup-form-submit-button" type="submit" value="Save" />
                                    <div className="devices-table-add-edit-popup-form-close-button" onClick={() => setShowEditDevicePopup(false)}>Close</div>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            )
        }

        if (showDeleteDevicePopup === true) {

            return (

                <div className="devices-table-add-edit-popup-container-overlay">
                    <div className="devices-table-add-edit-popup-container">
                        <div className="devices-table-add-edit-popup-container-top">
                            <div className="devices-table-add-edit-popup-closer" onClick={() => setShowDeleteDevicePopup(false)}>â¤«</div>
                            <div className="devices-table-add-edit-popup-title">Delete Device</div>
                        </div>
                        <div className="devices-table-add-edit-popup-container-middle">
                            <div className="devices-table-add-edit-popup-container-device-details">
                                <div className="devices-table-add-edit-popup-details-title">Device Details:</div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">System Name:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{deleteData.system_name}</div>
                                </div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">Type:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{deleteData.type}</div>
                                </div>
                                <div className="devices-table-add-edit-popup-container-device-details-values">
                                    <div className="devices-table-add-edit-popup-details-key">HDD Capacity:</div>
                                    <div className="devices-table-add-edit-popup-details-value">{deleteData.hdd_capacity}GB</div>
                                </div>
                            </div>
                            <div className="devices-table-add-edit-popup-confirm-text">Are you sure you wish to delete this device?</div>
                            <form className="devices-table-add-edit-popup-form-container" onSubmit={handleSubmit(onDeleteSubmit)}>
                                <div className="devices-table-add-edit-popup-container-bottom">
                                    <input className="devices-table-add-edit-popup-form-submit-button" type="submit" value="Confirm Delete" />
                                    <div className="devices-table-add-edit-popup-form-close-button" onClick={() => setShowDeleteDevicePopup(false)}>Close</div>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            )
        }
    }

    function sortData(sortType) {

        if (sortType === "STR") { return [...devicesData].sort((a, b) => a.system_name.localeCompare(b.system_name)); }
        if (sortType === "NUM") { return [...devicesData].sort((a, b) => parseInt(a.hdd_capacity) - parseInt(b.hdd_capacity)); }
    }

    function checkSortBy() {

        if (assignedSortSetting === "system_name") { 
            
            return <Table className="devices-table" columns={columns} data={sortData("STR")} /> 

        } else if (assignedSortSetting === "hdd_capacity") { 
            
            return <Table className="devices-table" columns={columns} data={sortData("NUM")} /> 

        } else { 
            
            return <Table className="devices-table" columns={columns} data={devicesData} />
        }
    }

    const columns = [
        {
            accessor: 'system_name',
            filterName: '',
            Filter: '',
            filter: '',
            Cell: (row) => <div className="devices-table-row-data-system-name">{row.value}</div>
        },
        {
            accessor: 'type',
            filterName: 'Device Type:',
            Filter: SelectColumnFilter,
            filter: 'includes',
            Cell: (row) => <div className="devices-table-row-data-type">{row.value}</div>
        },
        {
            accessor: 'hdd_capacity',
            filterName: '',
            Filter: '',
            filter: '',
            Cell: (row) => <div className="devices-table-row-data-hdd-capacity">{row.value}GB</div>
        },
        {
            accessor: 'edit_data',
            filterName: '',
            Filter: '',
            filter: '',
            Cell: (row) => <div className="devices-table-edit-device-button" onClick={() => editClickHandler(row.value)}>Edit</div>
        },
        {
            accessor: 'delete_data',
            filterName: '',
            Filter: '',
            filter: '',
            Cell: (row) => <div className="devices-table-delete-device-button" onClick={() => deleteClickHandler(row.value)}>Delete</div>
        }
    ];

    return (

        <div>
            {handleEditDeleteShow()}
            {checkSortBy()}
        </div>

    );
}

export default DevicesTable
