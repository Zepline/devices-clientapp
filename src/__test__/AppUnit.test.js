import { Component } from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { defaults } from '../config/index';
import logo from '../imgs/logo.png';
import DevicesTable from './DevicesTableUnit.test.js';
import restrictSpecialCharacters from '../components/helpers/string/restrictSpecialCharacters.js';
import axios from 'axios';

afterEach(cleanup);

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      tableSortSetting: this.props.sortSetting,
      showAddDevicePopup: false,
      system_name: '',
      type: '',
      hdd_capacity: ''
    }

    this.changeHandler = this.changeHandler.bind(this);
    this.submitHandler = this.submitHandler.bind(this);
  }

  changeHandler = (e) => {

    this.setState({ [e.target.id]: e.target.value });
  }

  submitHandler(e) {

    e.preventDefault();

    if (this.state.system_name.length !== 0) {

      if (defaults.types.includes(this.state.type)) {

        if (this.state.hdd_capacity >= 0 && !isNaN(this.state.hdd_capacity)) {

          const payload = {
            system_name: this.state.system_name,
            type: this.state.type,
            hdd_capacity: this.state.hdd_capacity
          }

          axios.post(defaults.deviceApiEndpoint, payload).then(() => {

            this.setState({
              showAddDevicePopup: false,
              systemName: '',
              type: '',
              hddCapacity: ''
            })

            setTimeout(function () { window.location.reload() }, 100);
          });
        }
      }
    }
  }

  updateTableSortSetting(event) {

    this.setState({ tableSortSetting: event.target.value });
  }

  showAddDevicePopup() {

    if (this.state.showAddDevicePopup === true) {

      return (

        <div data-testid="add-device-popup" className="devices-table-add-edit-popup-container-overlay">
          <div className="devices-table-add-edit-popup-container">
            <div className="devices-table-add-edit-popup-container-top">
              <div className="devices-table-add-edit-popup-closer" onClick={() => this.setState({ showAddDevicePopup: false })}>⤫</div>
              <div className="devices-table-add-edit-popup-title">Add Device</div>
            </div>
            <div className="devices-table-add-edit-popup-container-middle">
              <form className="devices-table-add-edit-popup-form-container" onSubmit={this.submitHandler}>
                <div className="devices-table-add-edit-popup-form-input-container">
                  <div className="devices-table-add-edit-devices-popup-input-title">System Name *</div>
                  <input className="devices-table-add-edit-devices-popup-input-data"
                    data-testid="system-name-input"
                    id="system_name"
                    type="text"
                    maxLength="50"
                    onKeyPress={(e) => restrictSpecialCharacters(e)}
                    value={this.state.system_name}
                    onChange={this.changeHandler}
                    required
                  />
                </div>
                <div className="devices-table-add-edit-popup-form-input-container">
                  <div className="devices-table-add-edit-devices-popup-input-title">Type *</div>
                  <select data-testid="type-input-form" className="devices-table-add-edit-devices-popup-input-select" id="type" value={this.state.type} onChange={this.changeHandler} required>
                    <option value="">Select Type</option>
                    <option value="Windows Workstation">Windows Workstation</option>
                    <option value="Windows Server">Windows Server</option>
                    <option value="Mac">Mac</option>
                  </select>
                </div>
                <div className="devices-table-add-edit-popup-form-input-container">
                  <div className="devices-table-add-edit-devices-popup-input-title">HDD Capacity (GB)*</div>
                  <input className="devices-table-add-edit-devices-popup-input-data"
                    data-testid="hdd-capacity-input"
                    id="hdd_capacity"
                    type="number"
                    min="0"
                    value={this.state.hdd_capacity}
                    onChange={this.changeHandler}
                    required
                  />
                </div>
                <div className="devices-table-add-edit-popup-container-bottom">
                  <input data-testid="submit-add-button" className="devices-table-add-edit-popup-form-submit-button" type="submit" value="Save" />
                  <div className="devices-table-add-edit-popup-form-close-button" onClick={() => this.setState({ showAddDevicePopup: false })}>Close</div>
                </div>
              </form>
            </div>
          </div>
        </div>
      )
    }
  }

  refreshDevicesTable() {

    return <DevicesTable tableSortSetting={this.state.tableSortSetting} />
  }

  render() {

    return (
      <div data-testid="root-container" className="root-container">
        <div className="sidebar-container"><img className="sidebar-icon-img" src={logo} alt="" /></div>
        <div className="App">
          <div className="devices-table-add-device-button-container">
            <div data-testid="add-device-button" className="devices-table-add-device-button" onClick={() => this.setState({ showAddDevicePopup: true })}>＋ Add Device</div>
          </div>
          <div className="devices-table-sort-by-container">
            <div className="devices-table-sort-filter-title">Sort by:</div>
            <select data-testid="select-sort-by-dropdown" className="devices-table-select-sort-filter" onChange={(setting) => this.updateTableSortSetting(setting)}>
              <option value="">None</option>
              <option value="system_name">System Name</option>
              <option value="hdd_capacity">HDD Capacity</option>
            </select>
          </div>
          {this.showAddDevicePopup()}
          {this.refreshDevicesTable()}
        </div>
      </div>
    )
  }
}

describe('Renders the App class component', () => {

  it("renders page without crashing", () => {

    render(<App />);
  })

  it("matches the snapshot", () => {

    const treeObject = renderer.create(<App />).toJSON();
    expect(treeObject).toMatchSnapshot();
  })
});

describe('Tests different components within the App class component', () => {

  it("clicks the add button and renders the add device popup", () => {

    render(<App />);

    userEvent.click(screen.getByTestId("add-device-button"));
    expect(screen.getByTestId("add-device-popup")).toBeInTheDocument();
  })

  it("selects the sortBy dropwdown option system name", () => {

    render(<App />);

    fireEvent.change(screen.getByTestId("select-sort-by-dropdown"), { target: { value: 'system_name' } })
    expect(screen.getByTestId("select-sort-by-dropdown")).toHaveTextContent("System Name");

    fireEvent.change(screen.getByTestId("select-sort-by-dropdown"), { target: { value: 'system_name' } })
    expect(screen.getByTestId("select-sort-by-dropdown")).toHaveTextContent("System Name");
  })

  it("selects the sortBy option hdd capacity", () => {

    render(<App />);

    fireEvent.change(screen.getByTestId("select-sort-by-dropdown"), { target: { value: 'hdd_capacity' } })
    expect(screen.getByTestId("select-sort-by-dropdown")).toHaveTextContent("HDD Capacity");
  })
});