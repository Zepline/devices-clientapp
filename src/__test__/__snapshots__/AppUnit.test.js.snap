// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the App class component matches the snapshot 1`] = `
<div
  className="root-container"
  data-testid="root-container"
>
  <div
    className="sidebar-container"
  >
    <img
      alt=""
      className="sidebar-icon-img"
      src="logo.png"
    />
  </div>
  <div
    className="App"
  >
    <div
      className="devices-table-add-device-button-container"
    >
      <div
        className="devices-table-add-device-button"
        data-testid="add-device-button"
        onClick={[Function]}
      >
        ï¼‹ Add Device
      </div>
    </div>
    <div
      className="devices-table-sort-by-container"
    >
      <div
        className="devices-table-sort-filter-title"
      >
        Sort by:
      </div>
      <select
        className="devices-table-select-sort-filter"
        data-testid="select-sort-by-dropdown"
        onChange={[Function]}
      >
        <option
          value=""
        >
          None
        </option>
        <option
          value="system_name"
        >
          System Name
        </option>
        <option
          value="hdd_capacity"
        >
          HDD Capacity
        </option>
      </select>
    </div>
    <div>
      <div
        data-testid="devices-table"
      >
        <div
          className="devices-table-container"
          role="table"
        >
          <div
            className="devices-table-header-container"
          >
            <div
              className="devices-table-filter-row-container"
              role="row"
            >
              <div
                className="devices-table-filter-container"
                colSpan={1}
                role="columnheader"
              >
                <div
                  className="devices-table-filter-title"
                >
                  
                </div>
                <div>
                  
                </div>
              </div>
              <div
                className="devices-table-filter-container"
                colSpan={1}
                role="columnheader"
              >
                <div
                  className="devices-table-filter-title"
                >
                  Device Type:
                </div>
                <div>
                  <select
                    className="devices-table-select-filter"
                    data-testid="type-filter"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      All
                    </option>
                    <option
                      value="Windows Server"
                    >
                      Windows Server
                    </option>
                  </select>
                </div>
              </div>
              <div
                className="devices-table-filter-container"
                colSpan={1}
                role="columnheader"
              >
                <div
                  className="devices-table-filter-title"
                >
                  
                </div>
                <div>
                  
                </div>
              </div>
              <div
                className="devices-table-filter-container"
                colSpan={1}
                role="columnheader"
              >
                <div
                  className="devices-table-filter-title"
                >
                  
                </div>
                <div>
                  
                </div>
              </div>
              <div
                className="devices-table-filter-container"
                colSpan={1}
                role="columnheader"
              >
                <div
                  className="devices-table-filter-title"
                >
                  
                </div>
                <div>
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="devices-table-body-container"
            role="rowgroup"
          >
            <div
              className="devices-table-row-container"
              role="row"
            >
              <div
                role="cell"
              >
                <div
                  className="devices-table-row-data-system-name"
                >
                  Abel-Server
                </div>
                 
              </div>
              <div
                role="cell"
              >
                <div
                  className="devices-table-row-data-type"
                >
                  Windows Server
                </div>
                 
              </div>
              <div
                role="cell"
              >
                <div
                  className="devices-table-row-data-hdd-capacity"
                >
                  100
                  GB
                </div>
                 
              </div>
              <div
                role="cell"
              >
                <div
                  className="devices-table-edit-device-button"
                  data-testid="devices-table-edit-button"
                  onClick={[Function]}
                >
                  Edit
                </div>
                 
              </div>
              <div
                role="cell"
              >
                <div
                  className="devices-table-delete-device-button"
                  data-testid="devices-table-delete-button"
                  onClick={[Function]}
                >
                  Delete
                </div>
                 
              </div>
            </div>
          </div>
        </div>
        <div
          className="pagination-container"
        >
          <div
            className="pagination-buttons-container"
          >
            <div
              className="pagination-button"
              disabled={true}
              onClick={[Function]}
            >
              &lt;&lt;
            </div>
            <div
              className="pagination-button"
              disabled={true}
              onClick={[Function]}
            >
              &lt;
            </div>
            <div
              className="pagination-button"
              disabled={true}
              onClick={[Function]}
            >
              &gt;
            </div>
            <div
              className="pagination-button"
              disabled={true}
              onClick={[Function]}
            >
              &gt;&gt;
            </div>
          </div>
          <div
            className="pagination-info-container"
          >
            <div
              className="page-of-text"
            >
              Page
               
              1
               of 
              1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders the table with data matches snapshot 1`] = `
<div>
  <div
    data-testid="devices-table"
  >
    <div
      className="devices-table-container"
      role="table"
    >
      <div
        className="devices-table-header-container"
      >
        <div
          className="devices-table-filter-row-container"
          role="row"
        >
          <div
            className="devices-table-filter-container"
            colSpan={1}
            role="columnheader"
          >
            <div
              className="devices-table-filter-title"
            >
              
            </div>
            <div>
              
            </div>
          </div>
          <div
            className="devices-table-filter-container"
            colSpan={1}
            role="columnheader"
          >
            <div
              className="devices-table-filter-title"
            >
              Device Type:
            </div>
            <div>
              <select
                className="devices-table-select-filter"
                data-testid="type-filter"
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  All
                </option>
              </select>
            </div>
          </div>
          <div
            className="devices-table-filter-container"
            colSpan={1}
            role="columnheader"
          >
            <div
              className="devices-table-filter-title"
            >
              
            </div>
            <div>
              
            </div>
          </div>
          <div
            className="devices-table-filter-container"
            colSpan={1}
            role="columnheader"
          >
            <div
              className="devices-table-filter-title"
            >
              
            </div>
            <div>
              
            </div>
          </div>
          <div
            className="devices-table-filter-container"
            colSpan={1}
            role="columnheader"
          >
            <div
              className="devices-table-filter-title"
            >
              
            </div>
            <div>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="devices-table-body-container"
        role="rowgroup"
      />
    </div>
    <div
      className="pagination-container"
    >
      <div
        className="pagination-buttons-container"
      >
        <div
          className="pagination-button"
          disabled={true}
          onClick={[Function]}
        >
          &lt;&lt;
        </div>
        <div
          className="pagination-button"
          disabled={true}
          onClick={[Function]}
        >
          &lt;
        </div>
        <div
          className="pagination-button"
          disabled={true}
          onClick={[Function]}
        >
          &gt;
        </div>
        <div
          className="pagination-button"
          disabled={true}
          onClick={[Function]}
        >
          &gt;&gt;
        </div>
      </div>
      <div
        className="pagination-info-container"
      >
        <div
          className="page-of-text"
        >
          Page
           
          1
           of 
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
